python/Ganga/GPIDev/Base/Proxy.py

```diff
@@ -229,24 +244,36 @@ def isType(_obj, type_or_seq):
         return isinstance(obj, bare_type_or_seq)
 
 def getName(_obj):
-    """Strip any proxy and then return an objects name"""
+    """Strip any proxy and then return an objects name
+    Args:
+        _obj (object): This may be an instance or a class"""
     obj = stripProxy(_obj)
     returnable = _getName(obj)
     return returnable
 
 def stripProxy(obj):
-    """Removes the proxy if there is one"""
+    """Removes the proxy if there is one
+    Args:
+        obj (object): This may be an instance or a class
+    """
     return getattr(obj, implRef, obj)
 
 
 def addProxy(obj):
-    """Adds a proxy to a GangaObject"""
+    """Adds a proxy to a GangaObject instance or class
+    Args:
+        obj (GangaObject, class): This may be a Ganga object or class which you're wanting to add a proxy to
```

Comment:
The reason that they're apparently inconsistent is that they do slightly different things. `getProxyClass()` will only return a proxy class if one has already been created for a type (i.e. it's a `GangaObject`) whereas `addProxy()` will most likely be creating a new proxy from scratch (and proxy reuse is primarily for caching).

I think it's unrealistic to not expect Ganga developers to have to understand the proxy system. I've worked to reduce the need to use it as much as possible but it can't be escaped entirely. Given that there's only a small handful of Ganga developers and the proxy system is such a core part of it I don't feel it's a big ask.

I agree that the naming isn't completely clear but I think it's certainly good enough for now.